### PR TITLE
test: install torch on cpu only

### PR DESCRIPTION
**Pull request by ThibaultFy**
## Related issue

`#` followed by the number of the issue

## Please check if the PR fulfills these requirements

- [ ] If the feature has an impact on the user experience, the [changelog](https://github.com/substra/substrafl/blob/main/CHANGELOG.md) has been updated
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The commit message follows the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification

### CODE REVIEW

File: tests/algorithms/pytorch/test_base_algo.py [hunk 1]
```diff
@@ -283,7 +284,11 @@ def test_rng_state_save_and_load(network, train_linear_nodes, session_dir, rng_s
     expected_output_round_2 = torch.rand(n_rng_sample)
 
     algo_deps = Dependency(
-        pypi_dependencies=["torch", "numpy"],
+        pypi_dependencies=[
+            "torch==2.0.1",
+            "numpy==1.24.3",
+            "--extra-index-url https://download.pytorch.org/whl/cpu",
+        ],
```
Comment (thbcmlowk) [inline tests/algorithms/pytorch/test_base_algo.py:291]: Wouldn't it be nice to use a fixture or at least a constant for this?

Comment (ThibaultFy) [reply tests/algorithms/pytorch/test_base_algo.py:291]: Yep you're right !

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

File: tests/algorithms/pytorch/test_base_algo.py (hunk 1 above)
Comment (thbcmlowk) [reply tests/algorithms/pytorch/test_base_algo.py]: Nice, thank you!

**Review by thbcmlowk — APPROVED** [review]

**Pull request merged**